### ISSUE TITLE
Fix accessibility violations

### ISSUE THREAD

**Issue by petermakowski** [canonical-web-and-design/maas-ui#3425]
#### Each issue contains:
- name and URL of the page
- violation level (e.g. critical)
- help link with potential fixes
- HTML node that caused the failure

- [x] "SSH keys for admin" page (/intro/user)
✖ 3 accessibility violations:

1. [CRITICAL] [aria-required-children] Certain ARIA roles must contain particular children:
   (https://dequeuniversity.com/rules/axe/4.3/aria-required-children?application=axeAPI)

- `<ul class="p-navigation__links" role="menu"></ul>`

2. [MODERATE] [page-has-heading-one] Page should contain a level-one heading:
   (https://dequeuniversity.com/rules/axe/4.3/page-has-heading-one?application=axeAPI)

- `<html lang="en">`

3. [MODERATE] [region] All page content should be contained by landmarks:
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- `<div class="p-status-bar"><div class="row"><div class="col-6"><strong data-testid="status-bar-maas-name">ubuntu-linux-20-04-desktop MAAS</strong>: <span data-testid="status-bar-version">3.1.0</span></div></div></div>`

- [x] "Devices" page (/devices)
  ✖ 5 accessibility violations:

1. [CRITICAL] [aria-required-parent] Certain ARIA roles must be contained by particular parents:
   (https://dequeuniversity.com/rules/axe/4.3/aria-required-parent?application=axeAPI)

- `<span role="columnheader" data-ng-click="sortTable('fqdn', 'devices')" data-ng-class="{'is-sorted': tabs.devices.predicate === 'fqdn', 'sort-asc': tabs.devices.reverse === false, 'sort-desc': tabs.devices.reverse === true}" class="is-sorted">
              <abbr>FQDN</abbr>
            </span>`

2. [CRITICAL] [button-name] Buttons must have discernible text:
   (https://dequeuniversity.com/rules/axe/4.3/button-name?application=axeAPI)

- `<button type="submit" class="p-search-box__button" alt="search">
          <i class="p-icon--search"></i>
        </button>`

3. [CRITICAL] [label] Form elements must have labels:
   (https://dequeuniversity.com/rules/axe/4.3/label?application=axeAPI)

- `<input type="checkbox" class="u-float--none" data-ng-click="toggleCheckAll('devices')" data-ng-checked="tabs.devices.allViewableChecked" id="check-all-devices" data-ng-disabled="hasActionsInProgress('devices')">`

4. [MODERATE] [page-has-heading-one] Page should contain a level-one heading:
   (https://dequeuniversity.com/rules/axe/4.3/page-has-heading-one?application=axeAPI)

- `<html lang="en">`

5. [MODERATE] [region] All page content should be contained by landmarks:
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- `<div id="status-bar"><div class="p-status-bar"><div class="row"><div class="col-6"><strong data-testid="status-bar-maas-name">ubuntu-linux-20-04-desktop MAAS</strong>: <span data-testid="status-bar-version">3.1.0</span></div></div></div></div>`

- [x] "Controllers" page (/controllers)
  ✖ 4 accessibility violations:
  
1. [CRITICAL] [aria-required-parent] Certain ARIA roles must be contained by particular parents:
   (https://dequeuniversity.com/rules/axe/4.3/aria-required-parent?application=axeAPI)

- `<span role="columnheader" data-ng-click="sortTable('fqdn', 'controllers')" data-ng-class="{'is-sorted': tabs.controllers.predicate === 'fqdn', 'sort-asc': tabs.controllers.reverse === false, 'sort-desc': tabs.controllers.reverse === true}" class="is-sorted">`

2. [CRITICAL] [label] Form elements must have labels:
   (https://dequeuniversity.com/rules/axe/4.3/label?application=axeAPI)

- `<input type="checkbox" class="u-float--none" data-ng-click="toggleCheckAll('controllers')" data-ng-checked="tabs.controllers.allViewableChecked" id="check-all-controllers" data-ng-disabled="hasActionsInProgress('controllers')">`

3. [MODERATE] [page-has-heading-one] Page should contain a level-one heading:
   (https://dequeuniversity.com/rules/axe/4.3/page-has-heading-one?application=axeAPI)

- `<html lang="en">`

4. [MODERATE] [region] All page content should be contained by landmarks:
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- `<div id="status-bar"><div class="p-status-bar"><div class="row"><div class="col-6"><strong data-testid="status-bar-maas-name">ubuntu-linux-20-04-desktop MAAS</strong>: <span data-testid="status-bar-version">3.1.0</span></div></div></div></div>`

- [x] "Subnets" page (/networks)
  ✖ 2 accessibility violations:
  
1. [MODERATE] [region] All page content should be contained by landmarks:
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- `<div id="status-bar"><div class="p-status-bar"><div class="row"><div class="col-6"><strong data-testid="status-bar-maas-name">ubuntu-linux-20-04-desktop MAAS</strong>: <span data-testid="status-bar-version">3.1.0</span></div></div></div></div>`

2. [CRITICAL] [select-name] Select element must have an accessible name:
   (https://dequeuniversity.com/rules/axe/4.3/select-name?application=axeAPI)

- `<select name="network-filter" class="p-form__control u-no-margin--bottom ng-pristine ng-untouched ng-valid ng-not-empty" id="networks-groupby" data-ng-model="groupBy" data-ng-change="updateGroupBy()">`

- [x] "Machines" page (/machines)
✖ 4 accessibility violations:
  
1. [CRITICAL] [aria-allowed-attr] Elements must only use allowed ARIA attributes:
   (https://dequeuniversity.com/rules/axe/4.3/aria-allowed-attr?application=axeAPI)

- `<a aria-selected="true" class="p-tabs__link" data-testid="tab-link-0 Machines" href="/MAAS/r/machines">0 Machines</a>`
- `<a aria-selected="false" class="p-tabs__link" data-testid="tab-link-1 Resource pool" href="/MAAS/r/pools">1 Resource pool</a>`

2. [CRITICAL] [label] Form elements must have labels:
   (https://dequeuniversity.com/rules/axe/4.3/label?application=axeAPI)

- `<input class="p-form-validation__input" id="D11d_iJmu-j63-BMqhS75" type="checkbox" aria-invalid="false" disabled="" data-testid="all-machines-checkbox">`

3. [MODERATE] [region] All page content should be contained by landmarks:
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- `<div class="section-header__main-row u-flex--between u-flex--wrap">`
- `<div class="p-status-bar"><div class="row"><div class="col-6"><strong data-testid="status-bar-maas-name">ubuntu-linux-20-04-desktop MAAS</strong>: <span data-testid="status-bar-version">3.1.0</span></div></div></div>`

4. [CRITICAL] [select-name] Select element must have an accessible name:
   (https://dequeuniversity.com/rules/axe/4.3/select-name?application=axeAPI)

- `<select class="p-form-validation__input" name="machine-groupings">`

- [x] "KVM" page (/kvm/lxd)
✖ 3 accessibility violations:
  
1. [CRITICAL] [aria-allowed-attr] Elements must only use allowed ARIA attributes:
   (https://dequeuniversity.com/rules/axe/4.3/aria-allowed-attr?application=axeAPI)

- `<a aria-selected="true" class="p-tabs__link" data-testid="lxd-tab" href="/MAAS/r/kvm/lxd">LXD</a>`
- `<a aria-selected="false" class="p-tabs__link" data-testid="virsh-tab" href="/MAAS/r/kvm/virsh">Virsh</a>`

2. [MODERATE] [heading-order] Heading levels should only increase by one:
   (https://dequeuniversity.com/rules/axe/4.3/heading-order?application=axeAPI)

- `<h4>No LXD hosts available</h4>`

3. [MODERATE] [region] All page content should be contained by landmarks:
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- `<div class="section-header__main-row u-flex--between u-flex--wrap">`
- `<div class="p-status-bar"><div class="row"><div class="col-6"><strong data-testid="status-bar-maas-name">ubuntu-linux-20-04-desktop MAAS</strong>: <span data-testid="status-bar-version">3.1.0</span></div></div></div>`

- [x] #3429
✖ 2 accessibility violations:
1. [MODERATE] [heading-order] Heading levels should only increase by one:
(https://dequeuniversity.com/rules/axe/4.3/heading-order?application=axeAPI)

- `<h4 data-testid="image-sync-text">Showing images synced from <strong>maas.io</strong></h4>`

2. [MODERATE] [region] All page content should be contained by landmarks:
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- `<div class="section__header-wrapper">`
- `<div class="p-status-bar"><div class="row"><div class="col-6"><strong data-testid="status-bar-maas-name">ubuntu-linux-20-04-desktop MAAS</strong>: <span data-testid="status-bar-version">3.1.0</span></div></div></div>`

- [x] "DNS" page (/domains)
✖ 1 accessibility violations:

1. [CRITICAL] [button-name] Buttons must have discernible text:
   (https://dequeuniversity.com/rules/axe/4.3/button-name?application=axeAPI)

- `<button class="p-button--base has-icon p-contextual-menu__toggle u-no-margin--bottom is-small is-dense" disabled="" aria-controls="UR4_MxcRqxcahZQdaAlVG" aria-expanded="false" aria-pressed="false" aria-haspopup="true" type="button">`

- [x] #3645
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- [ ] #3430
✖ 1 accessibility violation:

1. [MODERATE] [region] All page content should be contained by landmarks:
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- `<div class="section__header-wrapper">`
- `<div class="p-status-bar"><div class="row"><div class="col-6"><strong data-testid="status-bar-maas-name">ubuntu-linux-20-04-desktop MAAS</strong>: <span data-testid="status-bar-version">3.1.0</span></div></div></div>`

- [x] "Settings" page (/settings/configuration/general)
✖ 2 accessibility violations:
  
1. [MODERATE] [heading-order] Heading levels should only increase by one:
   (https://dequeuniversity.com/rules/axe/4.3/heading-order?application=axeAPI)

- `<h5 class="u-sv1">Data analytics</h5>`

2. [MODERATE] [region] All page content should be contained by landmarks:
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- `<div class="section__header-wrapper">`
- `<div class="p-status-bar"><div class="row"><div class="col-6"><strong data-testid="status-bar-maas-name">ubuntu-linux-20-04-desktop MAAS</strong>: <span data-testid="status-bar-version">3.1.0</span></div></div></div>`

- [x] "My preferences" page (/account/prefs/details)
✖ 1 accessibility violation:
  
1. [MODERATE] [region] All page content should be contained by landmarks:
   (https://dequeuniversity.com/rules/axe/4.3/region?application=axeAPI)

- `<div class="section__header-wrapper">`
- `<div class="p-status-bar"><div class="row"><div class="col-6"><strong data-testid="status-bar-maas-name">ubuntu-linux-20-04-desktop MAAS</strong>: <span data-testid="status-bar-version">3.1.0</span></div></div></div>`
